relations/src/r1cs/constraint_system.rs

```diff
@@ -309,91 +310,111 @@ impl<F: Field> ConstraintSystem<F> {
             return;
         }
 
-        let used_times = self.lc_num_times_used(true);
-
-        // step 2: Modify the lcs accordingly
-        let mut outlined_lcs = BTreeMap::new();
-
-        let mut additional_a_constraints = Vec::<LcIndex>::new();
-        let mut additional_c_constraints_witness_assignments = Vec::<usize>::new();
+        let mut num_times_used = self.lc_num_times_used(false);
 
+        let mut new_witness_linear_combinations = Vec::new();
+        let mut new_witness_indices = Vec::new();
+
+        // `new_lc_map` stores the modified linear combinations.
+        //
+        // This loop goes through all the LCs in the map, starting from
+        // the early ones, and decides whether or not to dedicate a witness
+        // variable for this LC.
+        //
+        // If true, the LC is replaced with 1 * this witness variable.
+        // Otherwise, the LC is inlined.
+        //
```

Comment:
```suggestion
        // `new_lc_map` stores the modified linear combinations.
        //
        // This loop goes through all the LCs in the map, starting from
        // the early ones, and decides whether or not to dedicate a witness
        // variable for this LC.
        //
        // If true, the LC is replaced with 1 * this witness variable.
        // Otherwise, the LC is inlined.
        //
        // Each iteration first updates the LC according to outlinings in prior iterations,
        // and then sees if it should be outlined, and if so adds the outlining to the map.
        //
```